### PR TITLE
Feature/add pathfactory fieldgroups v3

### PR DESCRIPTION
**Pull request by hitk1995**
Added Pathfactory custom fieldgroups for below 5 structures,
1] Pathfactory Session
2] Pathfactory Visitor
3] Pathfactory PageView
4] Pathfactory Content
5] Pathfactory Content Track

Fixes issue #1865

### CODE REVIEW

File: extensions/pathfactory/pathfactory-content-track.schema.json [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "meta:intendedToExtend": ["https://ns.adobe.com/xdm/context/experienceevent"],
+  "title": "PathFactory Content Track Fieldgroup Details - V1.1",
+  "type": "object",
+  "description": "PathFactory Content Track Schema Field",
+  "definitions": {
+    "pfContentTrackFields": {
+      "properties": {
+        "pf:pathfactoryContentTrack": {
```
Comment (diganttyagi1) [inline extensions/pathfactory/pathfactory-content-track.schema.json:19]: Please add a complete name - pathfactory in the namespace instead of abbreviations.

Comment (diganttyagi1) [reply extensions/pathfactory/pathfactory-content-track.schema.json:19]: @anandphatak

File: extensions/pathfactory/pathfactory-content-track.schema.json [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+          "type": "object",
+          "title": "Pathfactory Content Track",
+          "description": "Pathfactory Content Track Field Object",
+          "properties": {
+            "pf:createdAtUtc": {
+              "required": [],
+              "format": "date-time",
+              "type": "string",
+              "description": "",
```
Comment (diganttyagi1) [inline extensions/pathfactory/pathfactory-content-track.schema.json:28]: Please add Field descriptions for each field.

File: extensions/pathfactory/pathfactory-content-track.schema.json [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/xdm/mixins/pathfactory-contenttrack",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "meta:intendedToExtend": ["https://ns.adobe.com/xdm/context/experienceevent"],
+  "title": "PathFactory Content Track Fieldgroup Details - V1.1",
+  "type": "object",
+  "description": "PathFactory Content Track Schema Field",
```
Comment (diganttyagi1) [inline extensions/pathfactory/pathfactory-content-track.schema.json:15]: Please provide more detailed descriptions for this. What is the use case and usage of this field group

File: extensions/pathfactory/pathfactory-content-track.schema.json [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+  "definitions": {
+    "pfContentTrackFields": {
+      "properties": {
+        "pathfactory:contentTrack": {
+          "type": "object",
+          "title": "Pathfactory Content Track",
+          "description": "Pathfactory Content Track Field Object",
```
Comment (diganttyagi1) [inline extensions/pathfactory/pathfactory-content-track.schema.json:22]: Please provide more detailed descriptions for this.

File: extensions/pathfactory/pathfactory-content-track.schema.json [hunk 5]
```diff
@@ -0,0 +1,236 @@
+{
```
Comment (diganttyagi1) [inline extensions/pathfactory/pathfactory-content-track.schema.json:1]: Most of the descriptions needs to have more detailed descriptions for users to understand what these fields are and what are their use case / usage

File: extensions/pathfactory/pathfactory-content-track.schema.json [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+              "title": "Is Share Enabled",
+              "meta:xdmType": "boolean"
+            },
+            "pathfactory:isCaptureEnabled": {
+              "required": [],
+              "type": "boolean",
+              "description": "Boolean variable to check if capture is enabled in PathFactory",
+              "title": "Is Capture Enabled",
+              "meta:xdmType": "boolean"
+            },
```
Comment (diganttyagi1) [inline extensions/pathfactory/pathfactory-content-track.schema.json:138]: UUID, Seems all needs to be in uppercase

File: extensions/pathfactory/pathfactory-content-track.schema.json [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+              "meta:xdmType": "string"
+            },
+            "pathfactory:isCookieConsentEnabled": {
+              "required": [],
+              "type": "boolean",
+              "description": "Boolean variable to check if cookie consent is enabled in PathFactory",
+              "title": "Is Cookie Consent Enabled",
+              "meta:xdmType": "boolean"
+            },
+            "pathfactory:isGdprCookieConsentEnabled": {
```
Comment (diganttyagi1) [inline extensions/pathfactory/pathfactory-content-track.schema.json:153]: GDPR

File: extensions/pathfactory/pathfactory-content-track.schema.json [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+              "meta:xdmType": "boolean"
+            },
+            "pathfactory:targetContentOrder": {
+              "required": [],
+              "type": "string",
+              "description": "Target Content Order",
+              "title": "Target Content Order",
+              "meta:xdmType": "string"
+            },
+            "pathfactory:updatedAtUtc": {
```
Comment (diganttyagi1) [inline extensions/pathfactory/pathfactory-content-track.schema.json:216]: UTC

File: extensions/pathfactory/pathfactory-content-track.schema.json [hunk 9]
```diff
@@ -0,0 +1,236 @@
...
+        }
+      }
+    }
+  },
+  "allOf": [
+    {
+      "$ref": "#/definitions/pfContentTrackFields"
+    }
+  ],
+  "meta:status": "stable"
```
Comment (diganttyagi1) [inline extensions/pathfactory/pathfactory-content-track.schema.json:235]: Please meake the meta:status : experimental. We can mark that status stable only after testing on staging

File: extensions/pathfactory/pathfactory-content.schema.json [hunk 10]
```diff
@@ -0,0 +1,144 @@
+{
```
Comment (diganttyagi1) [inline extensions/pathfactory/pathfactory-content.schema.json:1]: Same comment : Please add more detailed descriptions to all fields

File: extensions/pathfactory/pathfactory-content.schema.json [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+          "description": "Pathfactory Content Field Object",
+          "required": [
+            "pathfactory:contentId",
+            "pathfactory:createdAtUtc",
+            "pathfactory:title",
+            "pathfactory:internalTitle"
+          ],
+          "type": "object",
+          "properties": {
+            "pathfactory:createdAtUtc": {
```
Comment (diganttyagi1) [inline extensions/pathfactory/pathfactory-content.schema.json:30]: UTC : All need to be uppercase

File: extensions/pathfactory/pathfactory-content.schema.json [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+            "pathfactory:contentId": {
+              "required": [],
+              "type": "integer",
+              "description": "Content ID",
+              "title": "Content ID",
+              "meta:xdmType": "int",
+              "minimum": -2147483648,
+              "maximum": 2147483647
+            },
+            "pathfactory:updatedAtUtc": {
```
Comment (diganttyagi1) [inline extensions/pathfactory/pathfactory-content.schema.json:47]: UTC uppercase

File: extensions/pathfactory/pathfactory-content.schema.json [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+              "meta:xdmType": "string"
+            },
+            "pathfactory:description": {
+              "required": [],
+              "type": "string",
+              "description": "Content Description",
+              "title": "Content Description",
+              "meta:xdmType": "string"
+            },
+            "pathfactory:sourceUrl": {
```
Comment (diganttyagi1) [inline extensions/pathfactory/pathfactory-content.schema.json:90]: URL : upper case

File: extensions/pathfactory/pathfactory-content.schema.json [hunk 14]
```diff
@@ -0,0 +1,144 @@
...
+        }
+      }
+    }
+  },
+  "allOf": [
+    {
+      "$ref": "#/definitions/pfContentFields"
+    }
+  ],
+  "meta:status": "stable"
```
Comment (diganttyagi1) [inline extensions/pathfactory/pathfactory-content.schema.json:143]: Update status to experimental

File: extensions/pathfactory/pathfactory-visitor.schema.json [hunk 15]
```diff
@@ -0,0 +1,130 @@
+{
```
Comment (diganttyagi1) [inline extensions/pathfactory/pathfactory-visitor.schema.json:1]: Please add more detailed descriptions

File: extensions/pathfactory/pathfactory-visitor.schema.json [hunk 16]
```diff
@@ -0,0 +1,130 @@
...
+              "meta:xdmType": "string"
+            },
+            "pathfactory:email": {
+              "required": [],
+              "type": "string",
+              "description": "Email Address",
+              "title": "Email",
+              "meta:xdmType": "string"
+            },
+            "pathfactory:visitorEcid": {
```
Comment (diganttyagi1) [inline extensions/pathfactory/pathfactory-visitor.schema.json:46]: ECID

File: extensions/pathfactory/pathfactory-visitor.schema.json [hunk 17]
```diff
@@ -0,0 +1,130 @@
...
+              "meta:xdmType": "date-time"
+            },
+            "pathfactory:visitorId": {
+              "required": [],
+              "type": "string",
+              "description": "Visitor ID",
+              "title": "Visitor ID",
+              "meta:xdmType": "string"
+            },
+            "pathfactory:visitorUuid": {
```
Comment (diganttyagi1) [inline extensions/pathfactory/pathfactory-visitor.schema.json:68]: UUID

File: extensions/pathfactory/pathfactory-visitor.schema.json [hunk 18]
```diff
@@ -0,0 +1,130 @@
...
+        }
+      }
+    }
+  },
+  "allOf": [
+    {
+      "$ref": "#/definitions/pfVisitorFields"
+    }
+  ],
+  "meta:status": "stable"
```
Comment (diganttyagi1) [inline extensions/pathfactory/pathfactory-visitor.schema.json:129]: Update status to experimental

File: extensions/pathfactory/pathfactory-content-track.schema.json [hunk 19]
```diff
@@ -0,0 +1,236 @@
...
+  "description": "PathFactory Content Track Schema Field",
+  "definitions": {
+    "pfContentTrackFields": {
+      "properties": {
+        "pathfactory:contentTrack": {
+          "type": "object",
+          "title": "Pathfactory Content Track",
+          "description": "Pathfactory Content Track Field Object",
+          "properties": {
+            "pathfactory:createdAtUtc": {
```
Comment (anandphatak) [inline extensions/pathfactory/pathfactory-content-track.schema.json:24]: All children node of `pathfactory:contentTrack` should not have `pathfactory:` namespace; please change them to `xdm:`. For example, `pathfactory:createdAtUtc` should be changed to `xdm:createdAtUTC` (notice the additional change for upper case UTC), and same for the rest of the siblings.

File: extensions/pathfactory/pathfactory-content-track.schema.json [hunk 20]
```diff
@@ -0,0 +1,236 @@
...
+              "required": [],
+              "type": "boolean",
+              "description": "Boolean variable to check is deleted",
+              "title": "Deleted",
+              "meta:xdmType": "boolean"
+            },
+            "pathfactory:experienceCustomUrl": {
+              "required": [],
+              "type": "string",
+              "description": "Experience Custom URL (PathFactory Experience)",
```
Comment (anandphatak) [inline extensions/pathfactory/pathfactory-content-track.schema.json:49]: Consider adding "format": "uri" if the values are of URI type so correct validations are performed during data ingestion. And change field name from `pathfactory:experienceCustomUrl` to `xdm:experienceCustomURL`

File: extensions/pathfactory/pathfactory-content-track.schema.json [hunk 21]
```diff
@@ -0,0 +1,236 @@
...
+          "properties": {
+            "pathfactory:createdAtUtc": {
+              "required": [],
+              "format": "date-time",
+              "type": "string",
+              "description": "Created Date in UTC format",
+              "title": "Created Date (UTC)",
+              "meta:xdmType": "date-time"
+            },
+            "pathfactory:appId": {
```
Comment (anandphatak) [inline extensions/pathfactory/pathfactory-content-track.schema.json:32]: All field name ending with Id should be capitalized; for example `pathfactory:appId` should be renamed to `xdm:appID`. Please refer https://github.com/adobe/xdm/blob/master/CONTRIBUTING.md#coding-styleguides for details.

File: extensions/pathfactory/pathfactory-content.schema.json [hunk 22]
```diff
@@ -0,0 +1,144 @@
...
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "meta:intendedToExtend": ["https://ns.adobe.com/xdm/context/experienceevent"],
+  "title": "PathFactory Content Fieldgroup Details - V1.1",
+  "type": "object",
+  "description": "PathFactory Content Schema Field",
+  "definitions": {
+    "pfContentFields": {
+      "properties": {
+        "pathfactory:content": {
```
Comment (anandphatak) [inline extensions/pathfactory/pathfactory-content.schema.json:19]: Same comments as earlier:

1. Change namespace for children of `pathfactory:content` to `xdm:`
2. Upper case for abbreviated names (Id to ID, Utc to UTC, Url to URL etc.)

**Review by anandphatak — COMMENTED** [review]
Please apply review comments to all files

File: extensions/pathfactory/pathfactory-content-track.schema.json [hunk 23]
```diff
@@ -0,0 +1,236 @@
...
+    "This work is licensed under a Creative Commons Attribution 4.0 International (CC BY 4.0) license",
+    "you may not use this file except in compliance with the License. You may obtain a copy",
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/xdm/mixins/pathfactory-contenttrack",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "meta:intendedToExtend": ["https://ns.adobe.com/xdm/context/experienceevent"],
+  "title": "PathFactory Content Track Fieldgroup Details - V1.1",
```
Comment (diganttyagi1) [inline extensions/pathfactory/pathfactory-content-track.schema.json:13]: We usually don't add version numbers in  the titles.

File: extensions/pathfactory/pathfactory-content-track.schema.json [hunk 24]
```diff
@@ -0,0 +1,236 @@
...
+  "type": "object",
+  "description": "This Schema Field is a data model within the PathFactory platform that captures and organizes metadata related to content tracks. It includes information such as track titles, content sequencing, and associated content assets. This schema is essential for managing how content is grouped, presented, and tracked within PathFactory, enabling personalized content journeys and providing insights into user engagement with specific content tracks.",
```
Comment (diganttyagi1) [inline extensions/pathfactory/pathfactory-content-track.schema.json:15]: This seems to be field group description. Would be good to reword this as 

"This Field Group captures and organises metadata related to content tracks of the PatFactory Platform...."

File: extensions/pathfactory/pathfactory-content.schema.json [hunk 25]
```diff
@@ -0,0 +1,144 @@
...
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/xdm/mixins/pathfactory-content",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "meta:intendedToExtend": ["https://ns.adobe.com/xdm/context/experienceevent"],
+  "title": "PathFactory Content Fieldgroup Details - V1.1",
+  "type": "object",
+  "description": "This Fields helps in organizing and storing key metadata for content assets. It includes details such as creation and update timestamps, titles, language, media type, and unique identifiers. This schema is essential for managing, categorizing, and delivering content effectively within PathFactory, ensuring each asset is easily identifiable and ready for personalized user experiences.",
```
Comment (diganttyagi1) [inline extensions/pathfactory/pathfactory-content.schema.json:15]: This Field Group ...

File: extensions/pathfactory/pathfactory-page-view.schema.json [hunk 26]
```diff
@@ -0,0 +1,652 @@
...
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/xdm/mixins/pathfactory-pageview",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "meta:intendedToExtend": ["https://ns.adobe.com/xdm/context/experienceevent"],
+  "title": "PathFactory Page Views Fieldgroup Details - V1.1",
+  "type": "object",
+  "description": "This schema field is a key element within the PathFactory platform's data schema, designed to capture and record detailed information about page views by visitors. It tracks every instance a visitor loads or reloads a page, providing crucial data for analyzing user engagement and content performance. The PathFactory Page Views Schema Field helps in understanding which pages are most frequently viewed, identifying popular content, and detecting patterns in visitor behavior. This data is instrumental in optimizing content delivery, personalizing visitor experiences, and driving strategic decisions based on real-time user interaction insights.",
```
Comment (diganttyagi1) [inline extensions/pathfactory/pathfactory-page-view.schema.json:15]: This Field Group provides fields to capture and record information ...

File: extensions/pathfactory/pathfactory-visitor.schema.json [hunk 27]
```diff
@@ -0,0 +1,130 @@
...
+    "This work is licensed under a Creative Commons Attribution 4.0 International (CC BY 4.0) license",
+    "you may not use this file except in compliance with the License. You may obtain a copy",
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/xdm/mixins/pathfactory-visitor",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "meta:intendedToExtend": ["https://ns.adobe.com/xdm/context/profile"],
+  "title": "PathFactory Visitor Fieldgroup Details - V1.1",
```
Comment (diganttyagi1) [inline extensions/pathfactory/pathfactory-visitor.schema.json:13]: Can we remove version number from title

File: extensions/pathfactory/pathfactory-session.schema.json [hunk 28]
```diff
@@ -0,0 +1,578 @@
...
+    "This work is licensed under a Creative Commons Attribution 4.0 International (CC BY 4.0) license",
+    "you may not use this file except in compliance with the License. You may obtain a copy",
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/xdm/mixins/pathfactory-session",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "meta:intendedToExtend": ["https://ns.adobe.com/xdm/context/experienceevent"],
+  "title": "PathFactory Session Fieldgroup Details - V1.1",
```
Comment (diganttyagi1) [inline extensions/pathfactory/pathfactory-session.schema.json:13]: Remove version number from title

File: extensions/pathfactory/pathfactory-page-view.schema.json [hunk 29]
```diff
@@ -0,0 +1,652 @@
...
+    "This work is licensed under a Creative Commons Attribution 4.0 International (CC BY 4.0) license",
+    "you may not use this file except in compliance with the License. You may obtain a copy",
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/xdm/mixins/pathfactory-pageview",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "meta:intendedToExtend": ["https://ns.adobe.com/xdm/context/experienceevent"],
+  "title": "PathFactory Page Views Fieldgroup Details - V1.1",
```
Comment (diganttyagi1) [inline extensions/pathfactory/pathfactory-page-view.schema.json:13]: Remove version number from title

File: extensions/pathfactory/pathfactory-content.schema.json [hunk 30]
```diff
@@ -0,0 +1,144 @@
...
+    "This work is licensed under a Creative Commons Attribution 4.0 International (CC BY 4.0) license",
+    "you may not use this file except in compliance with the License. You may obtain a copy",
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/xdm/mixins/pathfactory-content",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "meta:intendedToExtend": ["https://ns.adobe.com/xdm/context/experienceevent"],
+  "title": "PathFactory Content Fieldgroup Details - V1.1",
```
Comment (diganttyagi1) [inline extensions/pathfactory/pathfactory-content.schema.json:13]: Remove version number from title

**Review by anandphatak — APPROVED** [review]

**Comment by hitk1995** [comment]
@anandphatak @diganttyagi1 Will this be part of next XDM release ? or when will this PR be merged, can you please let us know ?
